### PR TITLE
add Content-Disposition header to archive

### PR DESCRIPTION
**Pull request by jrwren**

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1838/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1839/
Test FAILed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1840/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1841/
Test PASSed.

**Comment by bac** [comment]
👍

File: internal/v5/archive.go [hunk 1]
```diff
@@ -125,6 +125,7 @@ func (h *ReqHandler) SendEntityArchive(id *router.ResolvedURL, w http.ResponseWr
 	setArchiveCacheControl(w.Header(), h.isPublic(id))
 	header.Set(params.ContentHashHeader, blob.Hash)
 	header.Set(params.EntityIdHeader, id.PreferredURL().String())
+	header.Set("Content-Disposition", "attachment; filename="+id.PreferredURL().Name+".zip")
```
Comment (hatched) [inline internal/v5/archive.go:128]: Is there something somewhere restricting the file uploaded to be a .zip?

**Comment by hatched** [comment]
:+1: With one question
Thanks!

File: internal/v5/archive.go (hunk 1 above)
Comment (jrwren) [inline internal/v5/archive.go:128]: Yes. The spec.

**Comment by jrwren** [comment]
:shipit:

File: internal/v5/archive.go (hunk 1 above)
Comment (bac) [inline internal/v5/archive.go:128]: The Spec and 'charm push'?

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**